src/D2L.Bmx/ConsolePrompter.cs

```diff
@@ -49,28 +61,42 @@ internal class ConsolePrompter : IConsolePrompter {
 
 	string IConsolePrompter.PromptPassword() {
 		Console.Error.Write( "Okta Password: " );
-		var passwordChars = new List<char>();
 
-		while( true ) {
-			ConsoleKeyInfo input = Console.ReadKey( intercept: true );
-			if( input.Key == ConsoleKey.Enter ) {
-				Console.Error.Write( '\n' );
-				break;
-			}
-
-			if( input.Key == ConsoleKey.Backspace && passwordChars.Count > 0 ) {
-				passwordChars.RemoveAt( passwordChars.Count - 1 );
-			} else if( !char.IsControl( input.KeyChar ) ) {
-				passwordChars.Add( input.KeyChar );
+		if( RuntimeInformation.IsOSPlatform( OSPlatform.Windows ) ) {
+			var passwordChars = new List<char>();
+			while( true ) {
+				ConsoleKeyInfo input = Console.ReadKey( intercept: true );
+				if( input.Key == ConsoleKey.Enter ) {
+					Console.Error.Write( '\n' );
+					break;
+				}
+
+				if( input.Key == ConsoleKey.Backspace && passwordChars.Count > 0 ) {
+					passwordChars.RemoveAt( passwordChars.Count - 1 );
+				} else if( !char.IsControl( input.KeyChar ) ) {
+					passwordChars.Add( input.KeyChar );
+				}
 			}
+			return new string( passwordChars.ToArray() );
 		}
 
-		return new string( passwordChars.ToArray() );
+		string? password;
+		try {
+			// disable the terminal from echoing user input
+			using var stty = Process.Start( "stty", "-echo" );
+			stty.WaitForExit();
+			password = _stdinReader.ReadLine();
```

Comment:
This is similar to calling [tcsetattr](https://pubs.opengroup.org/onlinepubs/009696799/functions/tcsetattr.html), which a more proper implementation would do.
E.g. see Python's implementation of [getpass](https://docs.python.org/3/library/getpass.html):
https://github.com/python/cpython/blob/c39500db5254c9efee9937f951f58d87ee14cf2f/Lib/getpass.py#L68-L80

`stty` also calls `tcsetattr` under the hood, and is itself in the POSIX standard, so this should be fairly reliable, and a lot simpler than doing native library calls ourselves.

Reply:
Nice find

Reply:
Both.
I was looking for general advice on reading password from user input on POSIX systems, and I also knew Python had a getpass implementation that I'd like to check out. Both direction converged on using [termios](https://pubs.opengroup.org/onlinepubs/7908799/xsh/termios.h.html).
`stty` is often mentioned in answers to "how to read password in shell scripts", and checking the source code ([coreutils](https://github.com/wertarbyte/coreutils/blob/master/src/stty.c), [busybox](https://github.com/jgunthorpe/busybox/blob/master/coreutils/stty.c)), I see it simply calls termios functions under the hood, so decided to use this simpler method.

I did also look up calling native functions from .NET in the meantime ([docs](https://learn.microsoft.com/en-us/dotnet/standard/native-interop/)), but good cross-platform support (Mac & different distros of Linux) seems sketchy. There's also some boilerplate interop code (e.g. defining the termios struct) that's maybe too much for bmx.

There are also .NET libraries that help with calling libc (e.g. https://github.com/tmds/Tmds.LibC, https://github.com/KSBilodeau/TermiosLib), but they all use the older [`DllImportAttribute`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportattribute) not the newer [`LibraryImportAttribute`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.libraryimportattribute), and only the latter works with native AOT.